components/validation/src/main/java/org/wso2/carbon/identity/x509Certificate/validation/CertificateValidationUtil.java

```diff
@@ -1494,4 +1411,454 @@ private static Validator resourceToValidatorObject(
+
+        return processCertificateInStore(tenantDomain, certificate, null, false);
+    }
+
+    /**
+     * Update a certificate in the configuration store by certificate id.
+     *
+     * @param certificateId Certificate Id
+     * @param certificate X509Certificate
+     * @param tenantDomain Tenant Domain
+     * @return CertObject
+     * @throws CertificateValidationException Error when updating certificate
+     * @throws CertificateException Error when encoding certificate
+     * @throws CertificateMgtException Error when updating certificate
+     */
+    public static CertObject updateCertificateInConfigurationStoreByCertificateId(String certificateId,
+                                                                                  X509Certificate certificate,
+                                                                                  String tenantDomain)
+            throws CertificateValidationException, CertificateException, CertificateMgtException {
+
+        return processCertificateInStore(tenantDomain, certificate, certificateId, true);
+    }
+
+    /**
+     * Update the CA certificate in the certificate manager.
+     *
+     * @param certificateId Certificate id
+     * @param certificate X509Certificate
+     * @param tenantDomain Tenant domain
+     * @return X509Certificate
+     * @throws CertificateValidationException Error when updating certificate
+     */
+    public static X509Certificate updateCACertificateInCertificateManager(String certificateId,
+                                                                          X509Certificate certificate,
+                                                                          String tenantDomain)
+            throws CertificateValidationException {
+
+        try {
+            Certificate deletedCertificate = CertValidationDataHolder.getInstance()
+                    .getCertificateManagementService()
+                    .getCertificate(certificateId, tenantDomain);
+            CertValidationDataHolder.getInstance()
+                    .getCertificateManagementService()
+                    .updateCertificateContent(certificateId, encodeCertificate(certificate), tenantDomain);
+            log.debug("Successfully updated certificate with the id: " + certificateId + " in certificate manager.");
+            return decodeCertificate(deletedCertificate.getCertificateContent());
+        } catch (CertificateMgtException e) {
+            log.debug("Error when getting certificate from certificate manager.", e);
+            if (CertificateMgtErrors.ERROR_CERTIFICATE_DOES_NOT_EXIST.getCode().equals(e.getErrorCode())) {
+                throw new CertificateValidationException
+                        ("Certificate with the id: " + certificateId + " does not exist.", e);
+            } else {
+                throw new CertificateValidationException("Error when updating certificate in certificate manager.", e);
+            }
```

Comment:
`else` block is not needed

Reply:
fixed with https://github.com/wso2-extensions/identity-x509-commons/pull/57/commits/52f92488d265780c25c0d23c5ec638ca00bd3f59